tests/test_core.py

```diff
@@ -42,15 +42,15 @@ def test_locale_comparison():
     assert fi_FI != bad_en_US
 
 
-def test_can_return_default_locale(os_environ):
-    os_environ['LC_MESSAGES'] = 'fr_FR.UTF-8'
+def test_can_return_default_locale(monkeypatch):
+    monkeypatch.setenv('LC_MESSAGES', 'fr_FR.UTF-8')
```

Comment:
I think you can also use `monkeypatch.setenv` in `test_setuptools_frontend.py`:

https://github.com/python-babel/babel/blob/db4879136a7fbcef475f26b75dbdd65d0ce488f9/tests/messages/test_setuptools_frontend.py#L63-L64

i.e.

```diff
    if "BABEL_TOX_INI_DIR" in env:
+       monkeypatch.setenv("PYTHONPATH", env["BABEL_TOX_INI_DIR"])
-        env["PYTHONPATH"] = env["BABEL_TOX_INI_DIR"]

    # Initialize an empty catalog
    subprocess.check_call([
        sys.executable,
        "setup.py",
        "init_catalog",
        "-i", os.devnull,
        "-l", "fi",
        "-d", "inited",
+    ])
-    ], env=env)
```

Reply:
Good point, added!